### ISSUE TITLE
Export types from @aws-amplify/adapter-nextjs

### ISSUE THREAD

**Issue by jxdp** [aws-amplify/amplify-js#13039]
### Is this related to a new or existing framework?

Next.js

### Is this related to a new or existing API?

Authentication

### Describe the feature you'd like to request

The only export from `@aws-amplify/adapter-nextjs` is a function with the following signature:

```ts
createServerRunner(input: NextServer.CreateServerRunnerInput): NextServer.CreateServerRunnerOutput
```

I would like to be able to access the `NextServer.*` types, specifically `NextServer.Context`

### Describe the solution you'd like

Ideally, the `NextServer` type can just be exported. It would be a 1 line change with no drawbacks.

If that is too pervasive, then exporting just the context, something like:

```ts
export type NextServerContext = NextServer.Context;
```

### Describe alternatives you've considered

It is possible to get the types using typescript helpers in the following manner:

```ts
export type NextServerContext = NonNullable<
  Parameters<
    ReturnType<typeof createServerRunner>["runWithAmplifyServerContext"]
  >[0]["nextServerContext"]
>;

```

but this is somewhat fragile and puts more burden on the consumer and the already very overworked Typescript language server.

### Is this something that you'd be interested in working on?

- [x] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by jxdp** [comment]
> Hi @jxdp Thanks for opening the request. Your feedback is much appreciated. We had a consideration of enforcing the passing of required context parameters on every function call to reduce the chances of a wrong request or response object being used (e.g. the parameter object is created somewhere else and being reused), which further causes Amplify API calls to run in an incorrect context.
> 
> Could you explain further the use case that requires exporting this type?

Probably quite common a use-case of repackaging the server-side calls with the `serverRunner` singleton to make them easier to consume:

```ts
import { createServerRunner } from "@aws-amplify/adapter-nextjs";
import {
  fetchAuthSession,
  fetchUserAttributes,
  getCurrentUser,
} from "aws-amplify/auth/server";

import { resourcesConfig } from "./config";

type NextServerContext = NonNullable<
  Parameters<
    ReturnType<typeof createServerRunner>["runWithAmplifyServerContext"]
  >[0]["nextServerContext"]
>;

const serverRunner = createServerRunner({ config: resourcesConfig });

export async function getSession(context: NextServerContext) {
  return await serverRunner.runWithAmplifyServerContext({
    nextServerContext: context,
    operation: fetchAuthSession,
  });
}

export async function getUser(context: NextServerContext) {
  return await serverRunner.runWithAmplifyServerContext({
    nextServerContext: context,
    operation: getCurrentUser,
  });
}

export async function getUserAttributes(context: NextServerContext) {
  return await serverRunner.runWithAmplifyServerContext({
    nextServerContext: context,
    operation: fetchUserAttributes,
  });
}
```

**Comment by jxdp** [comment]
Closed by #13040
